heron/stmgr/src/cpp/manager/tmaster-client.cpp

```diff
@@ -217,5 +235,48 @@ void TMasterClient::SendHeartbeatRequest() {
   return;
 }
 
+void TMasterClient::SavedInstanceState(const proto::system::Instance& _instance,
+                                       const std::string& _checkpoint_id) {
+  proto::ckptmgr::InstanceStateStored message;
+  message.set_checkpoint_id(_checkpoint_id);
+  message.mutable_instance()->CopyFrom(_instance);
+  SendMessage(message);
+}
+
+void TMasterClient::SendRestoreTopologyStateResponse(proto::system::StatusCode _status,
+                                                     const std::string& _ckpt_id,
+                                                     sp_int64 _txid) {
+  proto::ckptmgr::RestoreTopologyStateResponse message;
+  message.mutable_status()->set_status(_status);
+  message.set_checkpoint_id(_ckpt_id);
+  message.set_restore_txid(_txid);
+  SendMessage(message);
+}
+
+void TMasterClient::HandleRestoreTopologyStateRequest(
+              proto::ckptmgr::RestoreTopologyStateRequest* _message) {
+  restore_topology_watch_(_message->checkpoint_id(), _message->restore_txid());
+  __global_protobuf_pool_release__(_message);
+}
+
+void TMasterClient::HandleStartStmgrStatefulProcessing(
```

Comment:
Why do we need this start_stateful_processing callback? Whether a topology is stateful or not is configured in user's topology code. Once a stmgr is started, it should already know if it's stateful or not by reading the topology config. And thus can behaves accordingly.